File: Documentation/4.0/Samples/java/src/test/java/net/ravendb/ClientApi/Session/HowTo/Refresh.java
package net.ravendb.ClientApi.Session.HowTo;

import net.ravendb.client.documents.DocumentStore;
import net.ravendb.client.documents.IDocumentStore;
import net.ravendb.client.documents.session.IDocumentSession;
import org.junit.Assert;

public class Refresh {
    private static class Employee {
        private String lastName;

        public String getLastName() {
            return lastName;
        }

        public void setLastName(String lastName) {
            this.lastName = lastName;
        }
    }

    public Refresh() {
        try (IDocumentStore store = new DocumentStore()) {
            try (IDocumentSession session = store.openSession()) {
                //region refresh_2
                Employee employee = session.load(Employee.class, "employees/1");
                Assert.assertEquals("Doe", employee.getLastName());

                // LastName changed to 'Shmoe'

                session.advanced().refresh(employee);

                Assert.assertEquals("Shmoe", employee.getLastName());
                //endregion
            }
        }
    }

    private interface IFoo {
        //region refresh_1
        <T> void refresh(T entity);
        //endregion
    }
}

File: Documentation/5.4/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/HowTo/Refresh.cs
using Raven.Client.Documents;
using Raven.Documentation.Samples.Orders;
using Xunit;

namespace Raven.Documentation.Samples.ClientApi.Session.HowTo
{
    public class Refresh
	{
		private interface IFoo
		{
			#region refresh_1
			void Refresh<T>(T entity);
			#endregion
		}

		public Refresh()
		{
			using (var store = new DocumentStore())
			{
				using (var session = store.OpenSession())
				{
					#region refresh_2
					Employee employee = session.Load<Employee>("employees/1");
					Assert.Equal("Doe", employee.LastName);

					// LastName changed to 'Shmoe'

					session.Advanced.Refresh(employee);
					Assert.Equal("Shmoe", employee.LastName);
					#endregion
				}
			}
		}
	}
}
